lib/file-source/src/file_server.rs

```diff
@@ -433,6 +441,28 @@ impl Fingerprinter {
     }
 }
 
+fn fingerprinter_read_until(mut r: impl Read, delim: u8, mut buf: &mut [u8]) -> io::Result<()> {
+    while !buf.is_empty() {
+        let read = match r.read(buf) {
+            Ok(0) => return Err(io::Error::new(io::ErrorKind::UnexpectedEof, "EOF reached")),
+            Ok(n) => n,
+            Err(ref e) if e.kind() == io::ErrorKind::Interrupted => continue,
+            Err(e) => return Err(e),
+        };
+
+        if let Some((pos, _)) = buf[..read].iter().enumerate().find(|(_, &c)| c == delim) {
+            for el in &mut buf[(pos + 1)..] {
+                *el = 0;
+            }
+            break;
+        }
+
+        let tmp = buf;
+        buf = &mut tmp[read..];
```

Comment:
Why `let tmp = buf;` need here?

Reply:
Turns you it's not needed here! It's a lifetime rebind trick, but it works fine without it. Good catch!